File: 452. Minimum Number of Arrows to Burst Balloons/solution.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        # edge case
        if not points:
            return 0
        
        # sort points
        points.sort(key = lambda x: (x[0], x[1]))
        
        # traverse all the points
        count = 0
        limit = sys.maxsize
        for i in range(len(points)):
            if points[i][0] > limit:
                count += 1
                limit = points[i][1]
            else:
                limit = min(limit, points[i][1])
                
        return count+1

File: 452. Minimum Number of Arrows to Burst Balloons/solution.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        if (points.length == 0)
            return 0;
        int arrows = 0;
        Arrays.sort(points, (a, b) -> a[0]==b[0]?(a[1]-b[1]):a[0]-b[0]);
        int minEnd = Integer.MIN_VALUE;
        for (int[] point: points) {
            int start = point[0];
            int end = point[1];
            if (start > minEnd) {
                minEnd = end;
                arrows++;
            } else {
                minEnd = Math.min(minEnd, end);
            }
        }
        if (minEnd == Integer.MIN_VALUE) {
            arrows++;
        }
        return arrows;
    }
}
